docs/adr/006_cups.md

```diff
@@ -0,0 +1,297 @@
+
+Only change is related to the concentrator device twin.
+
+The following "desired" properties should be properly set:
+
+```json
+"cups": {
+    "cupsUri": "https://IP_or_DNS:PORT",
+    "tcUri": "wss://IP_or_DNS:PORT",
+    "cupsCredCrc": INT,
+    "tcCredCrc": INT,
+    "cupsCredentialUrl": "https://...",
+    "tcCredentialUrl": "https://..."
+},
+// Following field to be used for first version client certificate validation
+"clientThumbprint": [ "Client certificate thumbprint" ]
+```
+
+- **'cupsCredCrc'**: should be computed as CRC32 checksum calculated over the concatenated credentials files `cups.{trust,cert,key}`
+- **'tcCredCrc'**: should be computed as CRC32 checksum calculated over the concatenated credentials files `tc.{trust,cert,key}`
+- **'cupsCredentialUrl'**: should point to the blob/secret containing the concatenated credentials `cups.{trust,cert,key}`
+- **'tcCredentialUrl'**: should point to the blob/secret containing the concatenated credentials `tc.{trust,cert,key}`
+- **'clientThumbprint'**: should include the thumbprint of the client certificate used for authenticating against the CUPS server. It is an array for allowing an old thumbprint to be accepted until the rotation mechanism completes
+
+### Facade Azure Function related changes
+
+The Azure Function should implement a new endpoint 'FetchConcentratorCredentials' which executes the following flow:
+
+```mermaid
+sequenceDiagram
+LoRaWanNetworkSrvModule->>Function: POST /fetchConcentratorCredentials
+Note right of LoRaWanNetworkSrvModule: Request includes 'stationEui' and 'credentialsType' (cups/tc/both)
+Function->>IoT Hub: Retrieve Concentrator device Twin
+Function->>Function: Parse *CredentialUrl string to understand credentials location
+alt *CredentialUrl is a KeyVault Secret
+    Function->>Azure KeyVault: Get KeyVault secret using Azure Function Managed Identity
+else *CredentialUrl is from Azure Blob Storage and no SAS is in Url
+    Function->>Azure Blob Storage: Get Blob using MSI
+else *CredentialUrl is plain Url
+    Function->>HTTPS Endpoint: Get Blob with HTTP GET Request
+end
+Function->>LoRaWanNetworkSrvModule: Return credential blob
+Note right of LoRaWanNetworkSrvModule: Response should be a json including base64 encoded credential
+
+```
+
+In addition to this, the Azure Function must be able to properly authenticate to target sink via Managed Identity (when using Azure Blob Storage or KeyVault).
+
+Considering that the default sink for the starter kit should be an Azure Storage Account, the template for the starter kit should be changed in such a way that a new Blob Container is created for uploading credential blob files.
+
+If using KeyVault secrets instead of Blob Storage, instructions on how to create a role assignment should be provided.  
+
+### LoRaWan Network Server changes
+
+- LoRaDeviceAPIServiceBase and its implementation (LoRaDeviceAPIService) should include a method for calling the 'FetchConcentratorCredentials' endpoint
```

Comment:
How do we provision the server certs of the LNS?

Reply:
We discussed this and afaik the plan is to stay with a cert on the file system as introduced in #734